### PR TITLE
Fix memory problems in flatten test

### PR DESCRIPTION
**Pull request by BradWhitlock**
I ran the Conduit tests after having built with the Intel compiler and the flatten test failed again in a new way. I ran through valgrind and it found out of bounds reads for the braid _quads_and_tris_ dataset. When I printed out that dataset, which gets used in flatten to make a 4x3 node mesh, it had some node ids that were larger than 11. The routine that made the connectivity was bad.

* I fixed the problem connectivity and now valgrind shows no errors in the flatten test
* I had to update a few partition baselines due to the connectivity changes
* I fixed some warnings

### CODE REVIEW

**Comment by BradWhitlock** [comment]
Wait a minute. I forgot to re-enable a test that started misbehaving.

File: src/libs/blueprint/conduit_blueprint_mesh_examples.cpp [hunk 1]
```diff
@@ -1820,16 +1818,14 @@ braid_quads_and_tris_offsets(index_t npts_x,
...
+        for(int32 i = 0; i < nele_x; i++)
         {
-            int32 n1 = yoff + i;
-            int32 n2 = n1 + (nele_x+1);
-            int32 n3 = n1 + 1 + (nele_x+1);
-            int32 n4 = n1 + 1;
+            int32 n1 = j * npts_x + i;
+            int32 n2 = j * npts_x + i + 1;
+            int32 n3 = (j+1) * npts_x + i + 1;
+            int32 n4 = (j+1) * npts_x + i;
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_examples.cpp:1829]: thanks for finding these mistakes

Comment (BradWhitlock) [reply src/libs/blueprint/conduit_blueprint_mesh_examples.cpp:1829]: Sure thing. It's all set now.

**Review by cyrush — APPROVED** [review]
thanks again!

**Pull request merged**